vars/jobStatusWrite.groovy

```diff
@@ -0,0 +1,25 @@
+// vars/jobStatusWrite.groovy
+
+/**
+ *
+ * Write out the job status for a job
+ *
+ */
+
+def call() {
+    if (!env.DAOS_STACK_JOB_STATUS_DIR) {
+        return
+    }
+    String job_name = env.JOB_NAME.replace('/', '_')
+    job_name += '_' + env.BUILD_NUMBER
+    String file_name = env.DAOS_STACK_JOB_STATUS_DIR + '/' + job_name
+
+    String job_status_text = writeYaml(data: jobStatus(),
+                                returnText: true)
+    // Need to use shell script for creating files that are not
+    // in the workspace.
+    job_status_text.split("\n").each { String line ->
+      sh(label: "Write jenkins_job_staus ${line}",
+         script: "echo '${line}' >> ${file_name}")
```

Comment:
Isn't this going to create a job step for every line written?  Can the whole YAML file not just be written all at once?

Reply:
Currently nothing is working other than it is not crashing from groovy syntax errors.
Having issue with determining the actual scope of a groovy global variable.